### PR TITLE
ListInAllNamespaces method added for statefulset pkg

### PR DESCRIPTION
**Pull request by yprokule**
Allows to list _StatefulSet_ resources in all the namespaces.

### CODE REVIEW

File: pkg/statefulset/list.go [hunk 1]
```diff
@@ -56,3 +56,47 @@ func List(apiClient *clients.Settings, nsname string, options ...metav1.ListOpti
 
 	return statefulsetObjects, nil
 }
+
+// ListInAllNamespaces returns statefulset inventory in the all namespaces.
```
Comment (achuzhoy) [inline pkg/statefulset/list.go:60]:
```suggestion
// ListInAllNamespaces returns statefulset inventory in all namespaces.
```

File: pkg/statefulset/statefulset_test.go [hunk 2]
```diff
@@ -1 +1,165 @@
...
+			expectedCount:          int(1),
+			statefulsetListOptions: []metav1.ListOptions{},
+		},
+		{
+			statefulsetMap: map[string][]string{
+				"one":   {"one-1", "one-2", "one-3"},
+				"two":   {"one-1", "two-2"},
+				"three": {"one-1"},
+			},
+			addToRuntimeObjects: true,
```
Comment (achuzhoy) [inline pkg/statefulset/statefulset_test.go:85]: Do you want to consider adding a test where addToRuntimeObjects is set to false?

File: pkg/statefulset/statefulset_test.go [hunk 3]
```diff
@@ -1 +1,165 @@
...
+			expectedCount:       int(3),
```
Comment (achuzhoy) [inline pkg/statefulset/statefulset_test.go:86]: Do we not expect an error here?
I see 6 entries

File: pkg/statefulset/list.go (hunk 1 above)
Comment (yprokule) [reply pkg/statefulset/list.go]: thanks, updated

File: pkg/statefulset/list.go [hunk 4]
```diff
@@ -56,3 +56,47 @@ func List(apiClient *clients.Settings, nsname string, options ...metav1.ListOpti
...
+		statefulsetBuilder := &Builder{
+			apiClient:  apiClient,
+			Object:     &copiedStatefulSet,
+			Definition: &copiedStatefulSet,
+		}
+
+		statefulsetObjects = append(statefulsetObjects, statefulsetBuilder)
+	}
+
+	glog.V(100).Infof("Found %d statefulsets across all the namespaces", len(statefulsetObjects))
```
[the remark this answers is not in the document]
Comment (yprokule) [reply pkg/statefulset/list.go]: thanks, updated

File: pkg/statefulset/statefulset_test.go (hunk 3 above)
Comment (yprokule) [reply pkg/statefulset/statefulset_test.go:86]: no, this one use label to match statefulsets with `demo=one-1` label (there are 3 of them)

File: pkg/statefulset/statefulset_test.go (hunk 2 above)
Comment (yprokule) [reply pkg/statefulset/statefulset_test.go:85]: That's what I was thinking about but haven't come up with any

Comment (trewest) [reply pkg/statefulset/statefulset_test.go:85]:
```
addToRuntimeObjects: false,
expectedCount: 0
```

File: pkg/statefulset/statefulset_test.go [hunk 5]
```diff
@@ -1 +1,165 @@
...
+	}{
+		{
+			statefulsetMap: map[string][]string{
+				"one":   {"one-1", "one-2", "one-3"},
+				"two":   {"two-1", "two-2"},
+				"three": {"three-1"},
+			},
+			addToRuntimeObjects:    true,
+			expectedCount:          int(6),
+			statefulsetListOptions: []metav1.ListOptions{},
```
[the remark this answers is not in the document]
Comment (yprokule) [reply pkg/statefulset/statefulset_test.go:69]: thanks, added

File: pkg/statefulset/statefulset_test.go (hunk 2 above)
Comment (yprokule) [reply pkg/statefulset/statefulset_test.go:85]: added, thanks

**Review by achuzhoy — APPROVED** [review]

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**